### PR TITLE
[action] add EvmTransaction to represent actions that run in EVM

### PR DESCRIPTION
**Pull request by dustinxie**
# Description
Please include a summary of the changes and the related issue. Please also include relevant motivation and context. List any dependencies that are required for this change.

Fixes #(issue)

## Type of change
Please delete options that are not relevant.
- [] Bug fix (non-breaking change which fixes an issue)
- [x] New feature (non-breaking change which adds functionality)
- [] Code refactor or improvement
- [] Breaking change (fix or feature that would cause a new or changed behavior of existing functionality)
- [] This change requires a documentation update

# How Has This Been Tested?
Please describe the tests that you ran to verify your changes. Provide instructions so we can reproduce. Please also list any relevant details for your test configuration
- [x] make test
- [] fullsync
- [] Other test (please specify)

**Test Configuration**:
- Firmware version:
- Hardware:
- Toolchain:
- SDK:

# Checklist:
- [] My code follows the style guidelines of this project
- [] I have performed a self-review of my code
- [] I have commented my code, particularly in hard-to-understand areas
- [] I have made corresponding changes to the documentation
- [] My changes generate no new warnings
- [] I have added tests that prove my fix is effective or that my feature works
- [] New and existing unit tests pass locally with my changes
- [] Any dependent changes have been merged and published in downstream modules

### CODE REVIEW

File: action/blob_tx.go [hunk 1]
```diff
@@ -0,0 +1,47 @@
...
+
+import (
+	"math/big"
+
+	"github.com/ethereum/go-ethereum/common"
+	"github.com/ethereum/go-ethereum/core/types"
+)
+
+// BlobTx represents an EIP-4844 transaction
+type BlobTx struct {
```
Comment (envestcc) [inline action/blob_tx.go:16]: how about the `blobVersionedHashes`?

**Review by envestcc — COMMENTED** [review]
blobTx could be introduced in another PR

File: action/protocol/execution/evm/evm.go [hunk 2]
```diff
@@ -200,8 +190,8 @@ func newParams(
 			GasPrice: execution.GasPrice(),
 		},
 		execution.Nonce(),
-		execution.Amount(),
```
Comment (envestcc) [inline action/protocol/execution/evm/evm.go:203]: why change to `Value()`

Comment (dustinxie) [reply action/protocol/execution/evm/evm.go:203]: for EVM part, to make our code as close to ether as possible

**Review by envestcc — APPROVED** [review]

File: action/protocol/execution/evm/evm.go [hunk 3]
```diff
@@ -121,21 +121,11 @@ func newParams(
...
-		contractAddrPointer *common.Address
-		getHashFn           vm.GetHashFunc
+		vmConfig  vm.Config
+		getHashFn vm.GetHashFunc
 	)
 
-	if dest := execution.Contract(); dest != action.EmptyAddress {
-		contract, err := address.FromString(execution.Contract())
-		if err != nil {
-			return nil, errors.Wrapf(err, "failed to decode contract address %s", dest)
```
Comment (millken) [inline action/protocol/execution/evm/evm.go:132]: new code panic instead of return error, no problem?

**Review by millken — APPROVED** [review]

**Comment by CoderZhi** [comment]
It is hard to see the value of representing Execution as EvmTransaction in this PR. It would be better if we can have some description explain the purpose and plan.

**Pull request merged**